### DISCUSSION TITLE
Not sure if I'm missing something, if the docs are confusing or if it's a bug or a mix of all of the above

### DISCUSSION THREAD

**Discussion by Edo78** [kool-dev/kool#481] [Q&A] [answered]
Hi, newbie here

I just discovered kool and it looks like something really nice so I start to check the doc for my specific use case: developing a vanilla node.js app.

The doc state "Stop installing different tech stacks on your local machine" so I'd give it a shot on a brand new linux machine with only the latest docker and docker compose installed. No node.js at all, nor global nor nvm, nothing.

Then I check the nodejs preset as documented [here](https://kool.dev/docs/presets/nodejs) but it didn't work. The `kool create nodejs my-project` create the `my-project` directory with:
```
$ ls
app.js  docker-compose.yml  kool.yml
```
but
```
$ kool run setup
Unable to find image 'kooldev/node:16' locally
16: Pulling from kooldev/node
7264a8db6415: Pull complete 
eee371b9ce3f: Pull complete 
93b3025fe103: Pull complete 
d9059661ce70: Pull complete 
88214d809a5a: Pull complete 
2d19fc4607dd: Pull complete 
ccf38205678e: Pull complete 
d31519ef8aa9: Pull complete 
f1e8d7630c51: Pull complete 
d9fb44626a33: Pull complete 
ece02a9cf269: Pull complete 
19ba0b6abfa0: Pull complete 
Digest: sha256:685de60f7bd6c74a326862aab70a9e8a267df1eb6ba05113fe2f2cb23ccf691f
Status: Downloaded newer image for kooldev/node:16
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /app/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/app/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in: /home/kool/.npm/_logs/2023-10-28T16_29_57_993Z-debug-0.log
exit status 254
```
Obviously the missing `package.json` is a big trouble for a node.js app
I can initialize the project with a `kool run npm init -f` and then I can run again `kool run setup`  without any errors but the `kool.yml` created by the preset is different from the one in the docs.
The docs show
```
scripts:
  node: kool exec app node
  npm: kool exec app npm # or yarn
  npx: kool exec app npx

  setup:
    - kool start
	# - add more setup commands
```
but the actual one is
```
# Here you can define shortcuts and aliases to common tasks (commands)
# you will run in your local environment or CI or deploy.
#  Use the scripts defined below with:
#    $ kool run <script>
# Learn more at: https://kool.dev/docs/getting-started/how-it-works#koolyml
scripts:
  node: kool exec app node
  npx: kool docker kooldev/node:16 npx
  setup:
    - kool run npm install
  reset:
    - kool run npm install
  # npm - helpers for JS handling
  npm: kool docker kooldev/node:16 npm
```
as you can see the actual setup just install the dependencies but it doesn't even try to start the app, to do that I have to run manually `kool start` and only then I can access `http://localhost:3000/` and see the wonderful `Hello world!` page.

If (and I know it's a really big IF) I understand correctly something it seems like the doc is about a previous version of the nodejs preset so it must be updated but the actual node.js preset isn't working by itself as the setup try to do an `npm install` without any `package.json` present and doesn't even try to start the container so it appears like a bug.

If I'm just missing something I need your help but if something is broken and someone can help me understand how things should works properly I'll be more than happy to write a PR to solve the issue.

**Comment by fabriciojs** [comment, marked as answer]
Hello @Edo78 !

I believe you have a good read of the situation.

We already have a few improvements ready to go with next version probably within this upcoming week - including updating the default Node version these presets come out the box with.

I will take a look and try to fix the preset by adding that init command so we have at least an empty package.json to start with.

I will refer the PR to this discussion once I get it up.

Thanks for trying out Kool, our main goal is facilitating the whole "web apps in a Docker container". Feedbacks like yours are most welcome and will help us improve where it's needed!

**Reply by fabriciojs** [reply]
PR is up - we should release this within a couple of days.

https://github.com/kool-dev/kool/pull/482

**Comment by Edo78** [comment]
Thank you very much :+1: 

A little OT, I just noticed that `presets/nodejs/config.yml` at line 6 contains `  - name: Creating new Laravel Application` maybe you should update that :wink:

**Reply by fabriciojs** [reply]
Thanks, got that fixed too.
